tests/explainers/test_tree.py

```diff
@@ -387,8 +387,9 @@ def test_catboost_categorical():
 
 
 def test_isolation_forest():
-    IsolationForest = pytest.importorskip("sklearn.ensemble.IsolationForest")
-    _average_path_length = pytest.importorskip("sklearn.ensemble._iforest._average_path_length")
+    from sklearn.ensemble import IsolationForest
+    from sklearn.ensemble._iforest import _average_path_length
```

Comment:
We should avoid importing private objects from 3rd party libraries: `_forest` is not part of sklearn's public API and could change or break without notice.

Would it be better to copy the relevant [private function](https://github.com/scikit-learn/scikit-learn/blob/150ff34e85f91a3a19a29dc4e19d50131fbb0251/sklearn/ensemble/_iforest.py#L531) and vendor it here?

Reply:
done